paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -595,9 +595,10 @@ internal class PaymentSheetViewModel @Inject internal constructor(
         )
     }
 
+    @Suppress("ComplexCondition")
     private fun paymentSelectionWithCvcIfEnabled(paymentSelection: PaymentSelection?): PaymentSelection? {
         return if (
-            isCvcRecollectionEnabled() &&
+            (isCvcRecollectionEnabled() || isCvcRecollectionForDeferred()) &&
```

Comment:
Are there tests for this?

Reply:
I think there's a bug here. Merchants could use the callback (intended to only be used with defferred intents) to affect non deferred flows.

Reply:
Yeah I was just thinking about that, is there a away to determine if it's a deferred configuration in BaseSheetViewModel?

Reply:
Updating so deferred checks that `(stripeIntent as? PaymentIntent) == null)` should fix it I believe

Reply:
I think we still have a payment intent, it's just got a null client secret. You'd have to manually test though.

Reply:
Updated to check the intent configuration is deferred. Removed the methods from `BaseSheetViewModel` and put in `PaymentSheetViewModel`. This functionality isn't needed in `PaymentOptionsViewModel` so there's no need for them to be in the base.